M2/cmake/FindGMP.cmake

```diff
@@ -54,9 +54,26 @@ macro(_gmp_check_version)
         ${GMP_MAJOR_VERSION}.${GMP_MINOR_VERSION}.${GMP_PATCHLEVEL_VERSION})
     endif()
   endforeach()
-
-  set(GMP_VERSION
-    ${GMP_MAJOR_VERSION}.${GMP_MINOR_VERSION}.${GMP_PATCHLEVEL_VERSION})
+  #If a direct search fails, use try_compile.
+  #This is for systems that put multiarch headers in other places.
+  if(NOT GMP_VERSION)
+    try_compile(GMP_TEST_SUCCESS ${CMAKE_BINARY_DIR} ${CMAKE_CURRENT_LIST_DIR}/gmp_version.c
```

Comment:
I think the code below could be significantly simplified if you use the constant `gmp_version` instead of the macros.

Also, it might be preferable to keep this self-contained (e.g. there are no .c sources in `share/cmake/Modules`), and in particular I think using `gmp_version` would make the source short enough that you could just use `file(WRITE ...)` to make a temporary file, compile it, and test the output.

Reply:
In particular, this source seems to work:
```c
#include "gmp.h"
#include "stdio.h"

void main() { printf("%s", gmp_version); }
```
and is short enough that it's reasonable to put it in `FindGMP.cmake` itself.

Reply:
The return type of `main` should be `int`, though.

Reply:
I've made this change, I didn't exactly know where to put the temporary file though, so it's currently in `${CMAKE_BINARY_DIR}`.

Reply:
That's fine, it already has a bunch of temporary files anyway. I think `${CMAKE_CACHEFILE_DIR}/CMakeFiles` is the correct place, to be pedantic, but it doesn't matter.